### PR TITLE
Todos

### LINKED ISSUE #56: check account has KYC
https://github.com/eoscostarica/proton-affiliate/blob/252f34416775444306afcf41829cf157cf14f189/contracts/affiliate/src/affiliate.cpp#L19-L24

---

###### This issue was generated by [todo](https://todo.jasonet.co) based on a `todo` comment in 252f34416775444306afcf41829cf157cf14f189 when #55 was merged. cc @eoscostarica.

### LINKED ISSUE #57: integrate with expiration_days from params table
https://github.com/eoscostarica/proton-affiliate/blob/252f34416775444306afcf41829cf157cf14f189/contracts/affiliate/src/affiliate.cpp#L98-L103

---

###### This issue was generated by [todo](https://todo.jasonet.co) based on a `todo` comment in 252f34416775444306afcf41829cf157cf14f189 when #55 was merged. cc @eoscostarica.

### PR DESCRIPTION
**Pull request by adriexnet**
### What does this PR do?

- Resolve #56
- Resolve #57

#### CheckList

- [x] Follow proper Markdown format
- [x] The content is adequate
- [x] The content is available in both english and spanish
- [x] I Ran a spell check

### CODE REVIEW

File: contracts/affiliate/src/affiliate.cpp [hunk 1]
```diff
@@ -17,7 +17,7 @@ ACTION affiliate::addadmin(name admin) {
 
   auto admin_itr = _users.find(admin.value);
   check(admin_itr == _users.end(), admin.to_string() + " account is already an affiliate");
-  // @todo: check account has KYC
+  check(has_valid_kyc(admin), "userinfo for " + admin.to_string() + " it's not verified");
```
Comment (xavier506) [inline contracts/affiliate/src/affiliate.cpp:20]: "KYC for " + admin.to_string() + " is not verified");

replace `it's` for `is `

`...is not verified`

 
 
 There is no need for an apostrophe here

example
**abbreviation**
it's time to go (abbreviation of it is)

**possessive**
It's main feature is performance 

feature belongs to something (le peretenece a algo)


https://www.grammarly.com/blog/its-vs-its/?&utm_source=google&utm_medium=cpc&utm_campaign=11907689446&utm_targetid=dsa-1233402314764&gclid=CjwKCAjwiLGGBhAqEiwAgq3q_mWTIJ02R83Z5sSZmGvlcJ2mKZQyG0p9eMZHD8iVCsdm9IoSA87BbBoCVg0QAvD_BwE&gclsrc=aw.ds

File: contracts/affiliate/src/affiliate.cpp [hunk 2]
```diff
@@ -46,7 +46,7 @@ ACTION affiliate::adduser(name admin, name user, uint8_t role) {
   check(admin != user, "cannot add self as user");
   check(is_account(user), user.to_string() + " account does not exist");
   check(role != user_roles::ADMIN, "role not allowed");
-  // @todo: check account has KYC
+  check(has_valid_kyc(user), "userinfo for " + user.to_string() + " it's not verified");
```
Comment (xavier506) [inline contracts/affiliate/src/affiliate.cpp:49]: "KYC for " + admin.to_string() + " is not verified");

File: contracts/affiliate/src/affiliate.cpp [hunk 3]
```diff
@@ -131,11 +132,7 @@ ACTION affiliate::verifykyc(name invitee) {
   check(_referral != _referrals.end(), "referral with invitee " + invitee.to_string() + " does not exist");
   check(_referral->status == referral_status::PENDING_KYC_VERIFICATION, "invalid status for invitee " + invitee.to_string() + " referral");
   check(_referral->expires_on > current_time_point(), "referral already expired for invitee " + invitee.to_string());
-
-  usersinfo_table _usersinfo(name("eosio.proton"), name("eosio.proton").value);
-  auto _userinfo = _usersinfo.find(invitee.value);
-  check(_userinfo != _usersinfo.end(), "userinfo for " + invitee.to_string() + " does not exist");
-  check(_userinfo->verified, "userinfo for " + invitee.to_string() + " it's not verified");
+  check(has_valid_kyc(invitee), "userinfo for " + invitee.to_string() + " it's not verified");
```
Comment (xavier506) [inline contracts/affiliate/src/affiliate.cpp:135]: KYC for "account" is not verified

**Review by xavier506 — CHANGES_REQUESTED** [review]
@adriexnet  awesome work with the kyc method.

please see my comments about some grammar errors

File: contracts/affiliate/src/affiliate.cpp (hunk 1 above)
Comment (adriexnet) [reply contracts/affiliate/src/affiliate.cpp]: @xavier506 thanks for the explication

**Review by xavier506 — APPROVED** [review]
LGTM

**Pull request merged**